### ISSUE TITLE
[HOLD for payment 2024-03-20] [$500] duplicate update available notifications

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36081]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** y
**Reproducible in production?:** not sure
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @arosiclair 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1707325061283629
 
## Action Performed:
1. Leave desktop app running
2. Wait for multiple staging app deploys
3. Go back online
## Expected Result: 
Only one "Update available" notification displays
## Actual Result: 
Multiple update notifications display

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Screenshot 2024-02-07 at 11 56 28 AM](https://github.com/Expensify/App/assets/38435837/e065b32e-666f-440c-823d-71ef46c04071)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b52a9d9f6e91e6ec</li>
        <li>Upwork Job ID: 1755982205531607040</li>
        <li>Last Price Increase: 2024-02-23</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

**Comment by slafortune** [comment]
@arosiclair should a C+ be added to this for review or is it something you will handle completely?

**Comment by arosiclair** [comment]
> @arosiclair should a C+ be added to this for review or is it something you will handle completely?

Yeah this will be external

**Comment by DylanDylann** [comment]
@arosiclair This issue relates to the notification system. So I think It should be handled on the BE side

**Comment by arosiclair** [comment]
The update available notifications are client side and should be fixable with [tags](https://developer.mozilla.org/en-US/docs/Web/API/Notification/tag). Not all notification issues are from the backend.

**Comment by DylanDylann** [comment]
Waiting proposals

**Comment by slafortune** [comment]
Checking with expert contributors

**Comment by arosiclair** [comment]
Still waiting on proposals

**Comment by arosiclair** [comment]
Thanks please leave a proposal when you're ready

**Comment by ShridharGoel** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When multiple updates happen on backend side, then frontend shows multiple notifications (one for each update).

### What is the root cause of that problem?
`pushUpdateAvailableNotification` method is called when a new update is available. It calls the `push` method which creates a new notification. No tag is being passed in the notification options which makes each notification as a unique notification. Below is the code.

https://github.com/Expensify/App/blob/e4a10d40ffb597bbcbd1e5c5f14c50899571194f/src/libs/Notification/LocalNotification/BrowserNotifications.ts#L121-L128

### What changes do you think we should make in order to solve the problem?
We should pass a tag `UpdateAvailable` in the `NotificationOptions`. When push method is called, the tag will be passed as an argument and then the same tag will be passed as an option while creating the Notification object.

**Comment by ShridharGoel** [comment]
@DylanDylann Can you have a look at my proposal?

**Comment by ShridharGoel** [comment]
@slafortune Hello, I'm new here and this looks like a good task to pick. Can my proposal be reviewed accordingly?

**Comment by Piotrfj** [comment]
Work on this issue is on hold due to higher priority task - regressions on BaseOptionsSelector. Will come back to this probably on monday

**Comment by arosiclair** [comment]
Yeah if you are already occupied @Piotrfj, let's go with another contributor. @ShridharGoel's proposal is already on the money so I'll assign them

**Comment by ShridharGoel** [comment]
Link to pull request: https://github.com/Expensify/App/pull/37595

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


Regression Test Proposal
1. Open the Expensify App.
2. Wait for a new deployment.
3. Check that "Update Available" notification is showing.
4. Wait for more deployments.
5. Only one update notification should show.


Do we agree 👍 or 👎

**Comment by slafortune** [comment]
@ShridharGoel offer sent [here](https://www.upwork.com/nx/wm/offer/101490548)
@DylanDylann Paid ✅

**Comment by slafortune** [comment]
All paid ✔️
